homeassistant/config.py

```diff
@@ -626,7 +626,7 @@ def _get_by_path(data: dict | list, items: list[str | int]) -> Any:
 
 
 def find_annotation(
-    config: dict | list, path: list[str | int]
+    config: dict | list, path: list[Hashable]
```

Comment:
It would be nice to note in the description that path isn't guaranteed to be passed a `str` or `int`, but than as I write that out, its seems self explanatory...